common/src/main/scala/uk/ac/wellcome/sqs/SQSReader.scala

```diff
@@ -39,4 +43,31 @@ class SQSReader @Inject()(sqsClient: AmazonSQS, sqsConfig: SQSConfig)
       .getMessages
       .toList
   }
+
+  private def processAndDeleteMessages(
+    messages: List[Message],
+    process: Message => Unit) = {
+      Future.sequence(messages.map { message =>
+        Future.fromTry(Try(process(message))
+          .recover {
+            case e: Throwable =>
+              error(s"Error processing message", e)
+              throw e
+        }).flatMap(_ => deleteMessage(message))
+      })
+  }
+
+  private def deleteMessage(message: Message) =
```

Comment:
I think it would be helpful to explain the SQS concepts in a comment somewhere. What should the processor do? Why do we delete a message after we’ve processed it? What happens if we don’t?

(I’m happy to write this if you want, but _somebody_ should write something.)